biomage/stage/stage.py

```diff
@@ -224,12 +263,307 @@ def create_manifest(templates, token):
+            experiment.split(" ")[0]
+            for experiment in prompt(questions)["experiments_to_stage"]
+        ]
+
+    else:
+
+        max_page = math.ceil(len(experiments) / experiments_per_page)
+        current_page = 0
+        done = False
+
+        while not done:
+
+            idx_start = experiments_per_page * current_page
+            idx_end = (current_page + 1) * experiments_per_page
+
+            choices = [
+                {
+                    "name": "{} {}".format(
+                        experiment["experimentId"].ljust(40),
+                        experiment["experimentName"],
+                    ),
+                    "checked": experiment["experimentId"] in chosen_experiments,
+                }
+                for experiment in experiments[idx_start:idx_end]
+            ]
+
+            # build choice options according to position in page
+            if current_page > 0:
+                choices = [{"name": previous_text}, *choices]
+
+            if current_page < max_page - 1:
+                choices = [*choices, {"name": next_text}]
+
+            choices = [*choices, {"name": done_text}]
+
+            questions = [
+                {
+                    "type": "checkbox",
+                    "name": "experiments_to_stage",
+                    "message": "Which experiments would you like to enable for the staging environment?\n"
+                    "Choose 'done' to exit",
+                    "choices": choices,
+                }
+            ]
+
+            answers = prompt(questions)["experiments_to_stage"]
+            page_action = None
+
+            # Check if any of the pagination options are selected
+            if previous_text in answers:
+                answers.remove(previous_text)
+                page_action = "previous"
+
+            if next_text in answers:
+                answers.remove(next_text)
+                page_action = "next"
+
+            if done_text in answers:
+                answers.remove(done_text)
+                page_action = "done"
+
+            chosen_experiments = set(
+                [
+                    *chosen_experiments,
+                    *[experiment.split(" ")[0] for experiment in answers],
+                ]
+            )
+
+            # Switch according to chosen action
+            if page_action == "next":
+                current_page += 1
+            elif page_action == "previous":
+                current_page -= 1
+            elif page_action == "done":
+                break
+
+    return chosen_experiments
+
+
+def check_file_etag(target, etag):
+    """
+    Copy s3 files in a bucket under a prefix
+    """
+    same_etag = False
+
+    try:
+        s3 = boto3.client("s3")
+        s3.head_object(Bucket=target["Bucket"], Key=target["Key"], IfMatch=etag)
+        same_etag = True
+    except ClientError as err:
+        if err.response['ResponseMetadata']['HTTPStatusCode'] not in [412, 404]:
```

Comment:
out of curiosity, what's this 412 precodition failed? I mean when does it happen? (add a comment if possible)

Reply:
HTTPError 412 is returned if the target's etag doesn't match with passed etag. This is a consequence of using `IfMatch` in `s3.head_object()`. I'll add it in as a comment,